### PR TITLE
folder_branch_ops: when journaling, only archive merged revisions

### PR DESCRIPTION
**Pull request by strib**
Issue: KBFS-1500

### CODE REVIEW

File: libkbfs/folder_branch_ops.go [hunk 1]
```diff
@@ -1978,8 +1978,21 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		return err
 	}
 
-	// Archive the old, unref'd blocks
-	fbo.fbm.archiveUnrefBlocks(irmd.ReadOnly())
+	// Archive the old, unref'd blocks if journaling is off.
+	doArchive := true
+	if jServer, err := GetJournalServer(fbo.config); err == nil {
```
Comment (akalin-keybase) [inline libkbfs/folder_branch_ops.go]: We do this in at least one other place, right? Maybe we should have a `TLFJournalEnabled(TlfID) bool` helper function

File: libkbfs/folder_branch_ops.go [hunk 2]
```diff
@@ -4669,6 +4683,30 @@ func (fbo *folderBranchOps) onTLFBranchChange(newBID BranchID) {
...
+	if bid != NullBranchID {
+		fbo.log.CDebugf(ctx, "Ignoring MD flush on branch %v for revision %d",
+			bid, rev)
+		return
+	}
+
+	fbo.log.CDebugf(ctx, "Archiving references for flushed MD revision %d", rev)
+
+	// Get that revision.
+	rmds, err := getMDRange(ctx, fbo.config, fbo.id(), NullBranchID,
```
Comment (akalin-keybase) [inline libkbfs/folder_branch_ops.go]: can use the `getSingleMD` helper from your other PR (or port that change into this one, or however you want to do it)

**Comment by akalin-keybase** [comment]
lgtm after comments and CI

File: libkbfs/folder_branch_ops.go (hunk 2 above)
Comment (strib) [inline libkbfs/folder_branch_ops.go]: I'll commit this one as is, and when I rebase the other one I'll fix it up.

**Pull request merged**